File: JavaScript Front End/src/Messages/CommandResponseMessages/GraphBuildingEdgeCreationFailureResponseMsg.js
const GraphBuildingFailureResponseMsg = require('./GraphBuildingFailureResponseMsg.js').GraphBuildingFailureResponseMsg;

class GraphBuildingEdgeCreationFailureResponseMsg extends GraphBuildingFailureResponseMsg
{
    constructor(jsonMessage)
    {
        super(jsonMessage);
        this.createdEdgeTemporaryId = this.jsonMessage["createdNodeTemporaryId"];
    }
}

module.exports = {GraphBuildingEdgeCreationFailureResponseMsg};

File: Java Back End/src/InfoSecCooker/Communication/Messages/CommandResponseMessages/GraphBuildingEdgeCreationFailureResponseMsg.java
package InfoSecCooker.Communication.Messages.CommandResponseMessages;

import InfoSecCooker.GraphLogic.GraphBuilding.Exceptions.GraphBuildingException;

public class GraphBuildingEdgeCreationFailureResponseMsg extends GraphBuildingFailureResponseMsg
{
    private long createdEdgeId;
    private long createdEdgeTemporaryId;

    public GraphBuildingEdgeCreationFailureResponseMsg(long createdEdgeId, GraphBuildingException exception, long createdEdgeTemporaryId, String additionalInfo)
    {
        super(exception, additionalInfo);
        this.createdEdgeId = createdEdgeId;
        this.createdEdgeTemporaryId = createdEdgeTemporaryId;
    }

    public GraphBuildingEdgeCreationFailureResponseMsg(long createdEdgeId, GraphBuildingException exception, long createdEdgeTemporaryId)
    {
        super(exception, "");
        this.createdEdgeId = createdEdgeId;
        this.createdEdgeTemporaryId = createdEdgeTemporaryId;
    }
}
